### DISCUSSION TITLE
Is there a way to pass command line params for browser instance in webdriverIO

### DISCUSSION THREAD

**Discussion by vibhu-shadwal-itv** [webdriverio/codemod#54] [Migration Issue] [answered]
So for my existing Protractor e2e test suites, i pass some params object variable through command line when running the tests, for different set of users, like - SUPER01_USER_EMAIL = browser.params.super01.email, which i can then pass through cli - protractor e2e/protractor.conf.js --baseUrl https://localhost:4200 --params.super01.email=****

Is there any easy way to replicate this in WebdriverIO?

thanks

**Comment by christian-bromann** [comment, marked as answer]
You can use environment variables or read CLI parameters in the wdio config add it to the config and read it through the [browser object](https://webdriver.io/docs/browserobject), e.g. `browser.config.myCustomConfig`.

**Reply by vibhu-shadwal-itv** [reply]
Thanks for the reply. But that would mean exposing the password in the config file then?

**Reply by christian-bromann** [reply]
If you want to pass in confidential information, better use environment variables.

**Reply by vibhu-shadwal-itv** [reply]
Thanks that helped. Another quick question for you. So we use protractor's browser.executeScript function to set localstorage values for chrome -
browser.executeScript('window.localStorage.setItem("token", "' + token + '");');
browser.executeScript('window.localStorage.setItem("lastAction", "' + Date.now() + '");');

which has been transformed by using codemod to 

browser.execute("window.localStorage.setItem("token", "' + token + '");");  -  but this seems to throw  error TS1005: ',' expected.
browser.execute('window.localStorage.setItem("lastAction", "' + Date.now() + '");');

Any suggestion how to fix it?

**Reply by christian-bromann** [reply]
You had a syntax error there, try:
```ts
browser.execute(
    (token) => window.localStorage.setItem("token", token),
    token
);

browser.execute(() => window.localStorage.setItem("lastAction", Date.now()));
```

**Reply by vibhu-shadwal-itv** [reply]
Another one for you if that's alright, so in few places of our protractor tests, we use cssContainingText locators where we pass the search text as a variable. codemod seems to error out while transforming it. Is there anyway in WebdriverIO to achieve it?? Thanks

few examples : 

element(by.cssContainingText("table tr td", SUPER_USER_EMAIL)).click();
const emails = element.all(by.cssContainingText("table tbody tr td", AE_USER_EMAIL));

**Reply by christian-bromann** [reply]
Just try: `await $('td=' + SUPER_USER_EMAIL).click()` as WebdriverIO supports [a similar locator](https://webdriver.io/docs/selectors#element-with-certain-text) too.
